src/table/boot.rs

```diff
@@ -236,6 +236,11 @@ impl BootServices {
         assert_eq!((self.stall)(time), Status::Success);
     }
 
+    /// Set the watchdog timer.
+    pub fn set_watchdog_timer(&self, timeout: usize, watchdog_code: u64, data_size: usize, watchdog_data: *mut u16) {
```

Comment:
This method should probably be unsafe since 1/it takes an unchecked pointer as a parameter and 2/data_size is not related in any way to the watchdog_data pointer and safety-critical caller mistakes can happen here as well.

Reply:
A safe alternative would be to pass in a slice of bytes.